src/EFCore.PG/Storage/Internal/Mapping/NpgsqlRangeTypeMapping.cs

```diff
@@ -47,5 +48,34 @@ static NpgsqlDbType GenerateNpgsqlDbType([NotNull] RelationalTypeMapping subtype
             p.DbType = subtypeMapping.DbType.Value;
             return NpgsqlDbType.Range | p.NpgsqlDbType;
         }
+
+        public override Expression GenerateCodeLiteral(object value)
+        {
+            var subtypeType = SubtypeMapping.ClrType;
+            var rangeType = typeof(NpgsqlRange<>).MakeGenericType(subtypeType);
+
+            var lower = rangeType.GetProperty(nameof(NpgsqlRange<int>.LowerBound)).GetValue(value);
+            var upper = rangeType.GetProperty(nameof(NpgsqlRange<int>.UpperBound)).GetValue(value);
+            var lowerInfinite = (bool)rangeType.GetProperty(nameof(NpgsqlRange<int>.LowerBoundInfinite)).GetValue(value);
+            var upperInfinite = (bool)rangeType.GetProperty(nameof(NpgsqlRange<int>.UpperBoundInfinite)).GetValue(value);
+            var lowerInclusive = (bool)rangeType.GetProperty(nameof(NpgsqlRange<int>.LowerBoundIsInclusive)).GetValue(value);
+            var upperInclusive = (bool)rangeType.GetProperty(nameof(NpgsqlRange<int>.UpperBoundIsInclusive)).GetValue(value);
```

Comment:
Should we cache or generate and cache some expression based accessors for these?

Reply:
Any reason? It is for migration code generation which doesn't require any performance improvements.

Reply:
As @YohDeadfall says we can do something, but as this is only migration code gen perf is not important at all... @austindrenski let me know if you're OK with this.

Reply:
Ok--scratch that then... Wasn't thinking about the actual application-time when I reviewed this.